### DISCUSSION TITLE
Roles and grouping

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#978] [Ideas]
I think it would be very beneficial to be able to group roles in the framework.  This feature should be easy to implement as in theory would have little impact on the current framework.

**Comment by hishamco** [comment]
There are roles built in Oqtane I am not sure what the benefit of grouping them

**Comment by leigh-pointer** [comment]
The benefit as i see it, is that a module could be installed and roles added to the framework for that module and identified by the group they are assigned.  Also quickly looking at the roles in the framework I would be clear where they are being utilized. Also give the data a little more clarity.

**Comment by sbwalker** [comment]
DNN had the concept of Role Groups.

 In large applications and websites the number of custom Roles can become cumbersome.  Role groups do not have any security implications of their own, they simply allow roles to be grouped for convenience in the user interface. For simplicity there should be a default role group that the existing internal roles are assigned to. 

Role management would include a filter to choose a specific role group or all role groups. The role editor component would need a way to assign a role group to a specific role. 

Permissions grid component would include a filter to choose a specific role group or all role groups.
